compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEDmaToCircularDma.cpp

```diff
@@ -0,0 +1,62 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree-amd-aie/IR/AMDAIEDialect.h"
+#include "iree-amd-aie/IR/AMDAIEOps.h"
+#include "iree-amd-aie/Transforms/Passes.h"
+
+#define DEBUG_TYPE "iree-amdaie-insert-circular-dma"
+
+namespace mlir::iree_compiler::AMDAIE {
+
+namespace {
+
+/// Convert dma copy operations to circular dma copy operations from L2 to L1
+/// and L1 to L2. This is hardcoded for now because of how MLIR-AIE is set up,
+/// but can be generalized in the future.
+LogicalResult convertDmaToCircularDma(Operation *op) {
+  IRRewriter rewriter(op->getContext());
+  op->walk([&](AMDAIE::DmaCpyNdOp dmaOp) {
+    auto sourceMemSpace = dmaOp.getSourceObjectFifo().getMemorySpace();
+    auto targetMemSpace = dmaOp.getTargetObjectFifo().getMemorySpace();
```

Comment:
`auto` -> `Attribute` ?

Reply:
Thanks for the catch, I copied some code from older code where I have more auto.. Updated it.